docs/doxygen/developer/style.txt

```diff
@@ -0,0 +1,294 @@
+
+Library file names should use one or two words to describe the functions implemented in that file. For example:
+- `core_mqtt_serializer.c`: Implements the MQTT library's packet serialization and deserialization functions.
+- `clock_posix.c`: Implements the platform clock component for POSIX systems.
+
+Declarations of internal functions, structures, macros, etc. of a library should be placed in a header file with an `_internal` suffix. The `_internal` header file should go in the `source/include/private` directory. For example:
+- `core_mqtt_internal.h`: Declares the MQTT library's internal functions, structures, macros, etc.
+
+<p>File names for demos should begin with `library_demo_`. 
+Unit tests must start with `library_` and end with `_utest`. 
+System tests must start with `library_` and end with `_system_test`. 
+The names should then specify the library being demoed or tested. 
+For example, the files names of the MQTT library's demos start with `mqtt_demo_`. 
+Additionally, test file names should describe what tests are implemented in the file, 
+such as `mqtt_demo_mutual_auth.c` for a file containing a demo that mutually 
+authenticates over TLS with an MQTT broker.</p>
+
+@subsection guide_developer_styleguide_naming_functions Functions (and function-like macros)
+@brief Naming convention of functions and function-like macros.
+
+@formattable{functions}
+@formattableentry{<b>Library</b>`_`<b>Description</b>,Externally-visible library function,`MQTT_Publish`}
+@formattableentry{<b>description</b>,`static` function (never uses `Aws` prefix),`sendPublish`<br>`eventCallback`}
+@formattableentry{`test_`<b>Library</b>`_`<b>accessedFunction_Descrption</b>,Test access function,`test_MQTT_ProcessLoop_Timer_Overflow`}
+
+Externally visible (i.e. not `static`) functions are [UpperCamelCased](https://en.wikipedia.org/wiki/Camel_case). Function names should then specify their library name, followed by an underscore, followed by a brief description of what the function does. For example:
+- `MQTT_SerializePublish`: This function is part of the public MQTT API. It serializes an MQTT PUBLISH packet into a buffer.
+
+Functions not visible outside their source file (i.e. `static` functions) have names that are [lowerCamelCased](https://en.wikipedia.org/wiki/Camel_case). These function names do not contain the library name or `Aws`. For example:
+- `receiveSingleIteration`: A `static` function in `core_mqtt.c`.
+
+@subsection guide_developer_styleguide_naming_types Types
+@brief Naming conventions of library `typedef` types.
+
+@formattable{types}
+@formattableentry{`Library`<b>Description</b>`_t`,General types in application-facing library header files,`MQTTStatus_t` (core_mqtt.h)}
+@formattableentry{`Library`<b>Function</b>`Info_t`,Application-facing parameter structure,`MQTTPublishInfo_t`<br>(Parameter structure to `MQTT_Publish`)}
```

Comment:
We also have `MQTTPacketInfo_t` and `MQTTDeserializedInfo_t`, which don't really fit with this. I'd probably just remove the second line